### DISCUSSION TITLE
isort for sorting import statements

### DISCUSSION THREAD

**Discussion by lucadellalib** [speechbrain/speechbrain#2361] [Feature Request]
### 🚀The feature

A standard good practice in Python is to group import statements in 3 sections separated by one blank line (see [https://peps.python.org/pep-0008/#imports](https://peps.python.org/pep-0008/#imports)):
1. Standard library imports
2. Third party imports
3. Local library imports

Furthermore, imports within each group should be sorted in lexicographically, with `import package`-style statements followed by `from package import ...`-style statements. These are the conventions followed by most top Python repositories (e.g. Pandas, TensorFlow, etc.). We should do the same in SpeechBrain to improve readability.

### Solution outline

`isort` ([https://github.com/PyCQA/isort](https://github.com/PyCQA/isort)) is a tool similar to `black` that automatically sorts import statements in all Python files. Just like `black` it can be customized and run as a `pre-commit` hook. Integration is straightforward and usually doesn't cause any compatibility issue. We should consider adding it to SpeechBrain.

**Comment by anautsch** [comment]
I like the idea, thanks for suggesting @lucadellalib ! 

There will be cases where manual sorting is needed though (@Adel-Moumen we discussed on the beamsearch refactoring). Referencing #1353 here, as we should put this one in too when we polish up our housekeeping for the next version.
